cardano-crypto-praos/cbits/crypto_vrf.c

```diff
@@ -40,51 +45,66 @@ crypto_vrf_primitive(void)
 int
 crypto_vrf_keypair(unsigned char *pk, unsigned char *sk)
 {
-    return crypto_vrf_ietfdraft03_keypair(pk, sk);
-}
+    unsigned char seed[32];
+    int           ret;
 
-int
-crypto_vrf_keypair_from_seed(unsigned char *pk, unsigned char *sk,
-			     const unsigned char *seed)
-{
-    return crypto_vrf_ietfdraft03_keypair_from_seed(pk, sk, seed);
+    randombytes_buf(seed, sizeof seed);
+    ret = crypto_vrf_seed_keypair(pk, sk, seed);
+    sodium_memzero(seed, sizeof seed);
+
+    return ret;
 }
 
 int
-crypto_vrf_is_valid_key(const unsigned char *pk)
+crypto_vrf_seed_keypair(unsigned char *pk, unsigned char *sk,
+                        const unsigned char *seed)
 {
-    return crypto_vrf_ietfdraft03_is_valid_key(pk);
+    ge25519_p3 A;
+
+    crypto_hash_sha512(sk, seed, 32);
+    sk[0] &= 248;
+    sk[31] &= 127;
+    sk[31] |= 64;
+
+    ge25519_scalarmult_base(&A, sk);
+    ge25519_p3_tobytes(pk, &A);
+
+    memmove(sk, seed, 32);
+    memmove(sk + 32, pk, 32);
+
+    return 0;
 }
 
-int
-crypto_vrf_prove(unsigned char *proof, const unsigned char *skpk,
-		 const unsigned char *m, const unsigned long long mlen)
+void
+crypto_vrf_sk_to_pk(unsigned char pk[crypto_vrf_PUBLICKEYBYTES],
+                    const unsigned char skpk[crypto_vrf_SECRETKEYBYTES])
 {
-    return crypto_vrf_ietfdraft03_prove(proof, skpk, m, mlen);
+    memmove(pk, skpk+32, 32);
 }
 
-int
-crypto_vrf_verify(unsigned char *output, const unsigned char *pk,
-		  const unsigned char *proof, const unsigned char *m,
-		  const unsigned long long mlen)
+void
+crypto_vrf_sk_to_seed(unsigned char seed[crypto_vrf_SEEDBYTES],
+                      const unsigned char skpk[crypto_vrf_SECRETKEYBYTES])
 {
-    return crypto_vrf_ietfdraft03_verify(output, pk, proof, m, mlen);
+    memmove(seed, skpk, 32);
 }
 
 int
-crypto_vrf_proof_to_hash(unsigned char *hash, const unsigned char *proof)
+crypto_vrf_prove(unsigned char *proof, const unsigned char *skpk,
+                 const unsigned char *m, const unsigned long long mlen)
 {
-    return crypto_vrf_ietfdraft03_proof_to_hash(hash, proof);
+    return crypto_vrf_ietfdraft13_prove(proof, skpk, m, mlen);
 }
 
-void
-crypto_vrf_sk_to_pk(unsigned char *pk, const unsigned char *skpk)
+int
+crypto_vrf_verify(unsigned char *output, const unsigned char *pk,
+                  const unsigned char *proof, const unsigned char *m,
+                  const unsigned long long mlen)
 {
-    crypto_vrf_ietfdraft03_sk_to_pk(pk, skpk);
+    return crypto_vrf_ietfdraft13_verify(output, pk, proof, m, mlen);
 }
 
-void
-crypto_vrf_sk_to_seed(unsigned char *seed, const unsigned char *skpk)
+int crypto_vrf_proof_to_hash(unsigned char *hash, const unsigned char *proof)
 {
-    crypto_vrf_ietfdraft03_sk_to_seed(seed, skpk);
-}
+    return crypto_vrf_ietfdraft13_proof_to_hash(hash, proof);
+}
```

Comment:
Include the new functions at the bottom, so that diff is simpler to read. and newline